### PR TITLE
10 Chat system homework #1 - Златина Лилова 

### PR DESCRIPTION
**Pull request by zlatililova**

### CODE REVIEW

**Comment by arnaudoff** [comment]
Please drop the .class and .idea files from the merge request.

File: homeworks/10/Chat-room/src/Server.java [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+            }catch (IOException e){
+                e.printStackTrace();
+            }
+
+        }
+    }
+
+    public static void main(String[] args){
+        Server server = new Server();
+        server.run();
```
Comment (LachezarLazarov) [inline homeworks/10/Chat-room/src/Server.java:150]: You don't need to run the server in a thread.

File: homeworks/10/Chat-room/src/Server.java [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+                        close();
+                    }else if (message.startsWith("/msg")){
+                        SimpleDateFormat formatDate = new SimpleDateFormat("hh:mm");
+                        String time = formatDate.format(new Date());
+                        String[] newMessage = message.split(" ", 2);
+                        sender = true;
+                        broadcastMessage("["+time+"] " + nickname + ": " + newMessage[1]);
+                        sender = false;
+                    }else{
+                    out.println("Wrong command!\n You can use the following commands\n" +
```
Comment (LachezarLazarov) [inline homeworks/10/Chat-room/src/Server.java:119]: You could use String.format()  or a single text block on multiple lines to make the code more readable.

File: homeworks/10/Chat-room/src/Server.java [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+            this.client = client;
+            sender = false;
+        }
+
+        @Override
+        public void run() {
+            try{
+                out = new PrintWriter(client.getOutputStream(), true);
+                in = new BufferedReader(new InputStreamReader(client.getInputStream()));
+                out.println("Welcome!\nYou can use the following commands\n" +
```
Comment (LachezarLazarov) [inline homeworks/10/Chat-room/src/Server.java:88]: You could use String.format() or a single text block on multiple lines to make the code more readable.

File: homeworks/10/Chat-room/src/Server.java [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+                while ((message = in.readLine()) != null){
+                    if(message.startsWith("/nick")){
+                        String[] newMessage = message.split(" ");
+                            broadcastMessage(nickname + " changed his nickname to " + newMessage[1]);
+                            nickname = newMessage[1];
+                    }else if (message.startsWith("/quit")) {
+                        SimpleDateFormat formatDate = new SimpleDateFormat("hh:mm");
+                        String time = formatDate.format(new Date());
+                        sender = true;
+                        broadcastMessage("[" + time + "] " + nickname + " left the chat!");
```
Comment (LachezarLazarov) [inline homeworks/10/Chat-room/src/Server.java:107]: You could use String.format() to make the code more readable.

File: homeworks/10/Chat-room/src/Server.java [hunk 5]
```diff
@@ -0,0 +1,152 @@
...
+                        sender = false;
+                        System.out.println(nickname + " left the chat");
+                        close();
+                    }else if (message.startsWith("/msg")){
+                        SimpleDateFormat formatDate = new SimpleDateFormat("hh:mm");
+                        String time = formatDate.format(new Date());
+                        String[] newMessage = message.split(" ", 2);
+                        sender = true;
+                        broadcastMessage("["+time+"] " + nickname + ": " + newMessage[1]);
```
Comment (LachezarLazarov) [inline homeworks/10/Chat-room/src/Server.java:116]: You could use String.format() to make the code more readable.

File: homeworks/10/Chat-room/src/Client.java [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+
+    public static void main(String[] args){
+        System.out.println("Welcome to our chat system! Type /connect <ip of chat server>:<port> to connect to chat room!");
+        String message;
+        BufferedReader input = new BufferedReader(new InputStreamReader(System.in));
+        String clientIp = null;
+        int clientPort = 0;
+        try {
+            while ((message = input.readLine()) != null){
+                if(message.startsWith("/connect")){
```
Comment (LachezarLazarov) [inline homeworks/10/Chat-room/src/Client.java:82]: You should aadd error handling when a user sends an invalid ip:port combination.

File: homeworks/10/Chat-room/src/Client.java [hunk 7]
```diff
@@ -0,0 +1,101 @@
...
+    class InputHandler implements Runnable{
+
+        @Override
+        public void run() {
+               try{
+                BufferedReader input = new BufferedReader(new InputStreamReader(System.in));
+                while(!done){
+                    String message = input.readLine();
+                    out.println(message);
+                    if(message.equals("/quit")){
```
Comment (LachezarLazarov) [inline homeworks/10/Chat-room/src/Client.java:63]: I suggest changing the "/quit" command to disconnect the client from the room isntead of ending the process.

**Review by LachezarLazarov — COMMENTED** [review]
Your overall grade is 5.50. Your code is mostly readable but there are some small issues I found.

**Pull request merged**